engine/api/secret/secretbackend/types.go

```diff
@@ -0,0 +1,75 @@
+package secretbackend
+
+import "encoding/gob"
+
+func init() {
+	gob.Register(Options{})
+	gob.Register(Secrets{})
+	gob.Register(SecretError(""))
+}
+
+//Driver is a plugin interface for retrieve CDS secrets
+type Driver interface {
+	Init(MapVar) error
+	Name() string
+	GetSecrets() Secrets
```

Comment:
go-plugin seems to be quite unstable with double return values
one of this days i should go through protobuf and grpc...